src/pycram/process_modules/icub_process_modules.py

```diff
@@ -0,0 +1,686 @@
+    pass
+
+def update_hand(hand_values,ctp_port):
+    yarp_bottle_msg: yarp.Bottle = yarp.Bottle()
+    yarp_bottle_reply: yarp.Bottle = yarp.Bottle()
+    yarp_bottle_msg.addVocab32('c', 't', 'p', 'q')
+    yarp_bottle_msg.addVocab32('t', 'i', 'm', 'e')
+    yarp_bottle_msg.addFloat32(1.5)
+    yarp_bottle_msg.addVocab32('o', 'f', 'f')
+    yarp_bottle_msg.addInt32(7)
+    yarp_bottle_msg.addVocab32('p', 'o', 's')
+    target_loc: yarp.Bottle = yarp_bottle_msg.addList()
+    for i in hand_values:
+        target_loc.addFloat32(i)
+
+
+    ctp_port.write(yarp_bottle_msg, yarp_bottle_reply)
+    reply_vocab = yarp_bottle_reply.get(0).asVocab32()
+
+    if reply_vocab == NO_ACK_VOCAB:
+        loginfo("Received: NO_ACK")
+        return False
+    elif reply_vocab == ACK_VOCAB:
+        loginfo("Received: ACK")
+        return True
+    else:
+        loginfo("Received: another reply")
+        return False
+
+def update_part(state_port,ctp_port, joint_to_change_idx, joints_to_change_pos):
+    if len(joint_to_change_idx):
+        part_state: yarp.Bottle = state_port.read(shouldWait=True)
+        part_new_states = []
+        for i in range(part_state.size()):
+            part_new_states.append(part_state.get(i).asFloat32())
+
+        for i in range(len(joint_to_change_idx)):
+            part_new_states[joint_to_change_idx[i]] = math.degrees(joints_to_change_pos[i])
+
+        yarp_bottle_msg: yarp.Bottle = yarp.Bottle()
+        yarp_bottle_reply: yarp.Bottle = yarp.Bottle()
+        yarp_bottle_msg.addVocab32('c', 't', 'p', 'q')
+        yarp_bottle_msg.addVocab32('t', 'i', 'm', 'e')
+        yarp_bottle_msg.addFloat32(1.5)
+        yarp_bottle_msg.addVocab32('o', 'f', 'f')
+        yarp_bottle_msg.addInt32(0)
+        yarp_bottle_msg.addVocab32('p', 'o', 's')
+        target_loc: yarp.Bottle = yarp_bottle_msg.addList()
+        for i in part_new_states:
+            target_loc.addFloat32(i)
+
+        ctp_port.write(yarp_bottle_msg, yarp_bottle_reply)
+        reply_vocab = yarp_bottle_reply.get(0).asVocab32()
+
+        if reply_vocab == NO_ACK_VOCAB:
+            loginfo("Received: NO_ACK")
+            return False
+        elif reply_vocab == ACK_VOCAB:
+            loginfo("Received: ACK")
+            return True
+        else:
+            loginfo("Received: another reply")
+            return False
+
+HAND_CLOSED = [60.0,60.0,0,85.0,20,85,20,85,85]
+HAND_OPENED = [0   ,0   ,0,0   ,0 ,0 ,0 ,0 ,0]
+class iCubNavigationReal(ProcessModule):
+    """
+    The process module to move the robot from one position to another.
+    """
+
+    def _execute(self, desig: MoveMotion):
+        loginfo("iCubNavigationReal function. No implementation ATM")
+
+
+class iCubMoveHeadReal(ProcessModule):
+    """
+        This process module moves the head to look at a specific point in the world coordinate frame.
+        This point can either be a position or an object.
+        """
+
+    def __init__(self,lock,cmd_port:yarp.RpcClient):
```

Comment:
The type annotation yarp.RpcClient will not work if yarp is not there. You can put it in quotation marks to prevent the interpreter evaluating it